lib/internal/bootstrap/realm.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+/**
+ * This file is executed in every realm that is created by Node.js, including
```

Comment:
Files in `lib/internal/per_context` and the file `loaders.js` can not load other built-in modules as the loader is not initialized yet -- `BuiltinLoader::CompileAndCall` gets per realms handles and calls into the built-in modules.

As this file is loaded when building the built-in snapshot, I think the overhead at bootstrapping should be negligible:

```
                                                                                     confidence improvement accuracy (*)   (**)  (***)
misc/startup.js count=30 mode='process' script='benchmark/fixtures/require-builtins'                 0.14 %       ±1.08% ±1.43% ±1.87%
misc/startup.js count=30 mode='process' script='test/fixtures/semicolon'                             0.35 %       ±0.94% ±1.25% ±1.62%
misc/startup.js count=30 mode='worker' script='benchmark/fixtures/require-builtins'                  0.30 %       ±1.20% ±1.60% ±2.08%
misc/startup.js count=30 mode='worker' script='test/fixtures/semicolon'                             -0.24 %       ±1.28% ±1.70% ±2.21%
```

Reply:
Thanks for the elaboration. I think this can be rebased on top of https://github.com/nodejs/node/pull/47215 to avoid using built-in loaders before they are properly installed in the realm's handle slots.